zerver/lib/soft_deactivation.py

```diff
@@ -33,3 +37,24 @@ def do_soft_deactivate_users(users):
             )
             realm_logs.append(log)
         RealmAuditLog.objects.bulk_create(realm_logs)
+
+def get_users_for_soft_deactivation(realm, inactive_for_days):
+    # type: (Realm, int) -> List[UserProfile]
+    users_activity = list(UserActivity.objects.filter(
+        user_profile__realm=realm,
+        user_profile__is_bot=False,
+        user_profile__long_term_idle=False).values(
```

Comment:
This should also be filtering on `is_active=True`; if the user isn't active, they shouldn't be soft-deactivated.